commit 46d384d476b9d1381498c7a91ac95a5c3f566611
Author: Dmitry Ivankov <divanorama@gmail.com>
Date:   Sun Jan 31 23:57:47 2021 +0100

    Patch grpc v1.32.0 for Bazel build on Darwin NixOS
    
    Remove thread local use in re2 on darwin

diff --git a/BUILD b/BUILD
index d5e98d6e13..b2d71f69ac 100644
--- a/BUILD
+++ b/BUILD
@@ -19,6 +19,7 @@ licenses(["notice"])
 exports_files([
     "LICENSE",
     "etc/roots.pem",
+    "re2_no_thread_local_darwin.patch",
 ])
 
 package(
diff --git a/bazel/grpc_deps.bzl b/bazel/grpc_deps.bzl
index 1f7794eb28..709ef3e564 100644
--- a/bazel/grpc_deps.bzl
+++ b/bazel/grpc_deps.bzl
@@ -216,6 +216,8 @@ def grpc_deps():
     if "com_github_google_re2" not in native.existing_rules():
         http_archive(
             name = "com_github_google_re2",
+	    patch_args = ["-p1"],
+	    patches = ["@com_github_grpc_grpc//:re2_no_thread_local_darwin.patch"],
             sha256 = "9f385e146410a8150b6f4cb1a57eab7ec806ced48d427554b1e754877ff26c3e",
             strip_prefix = "re2-aecba11114cf1fac5497aeb844b6966106de3eb6",
             urls = [
diff --git a/re2_no_thread_local_darwin.patch b/re2_no_thread_local_darwin.patch
new file mode 100644
index 0000000000..9016d29073
--- /dev/null
+++ b/re2_no_thread_local_darwin.patch
@@ -0,0 +1,19 @@
+commit d87478cb3b84eba23ae60aa45d89758c46802f99
+Author: Dmitry Ivankov <divanorama@gmail.com>
+Date:   Sun Jan 31 23:50:37 2021 +0100
+
+    Workaround bazel 4.0.0 build on Darwin on NixOS
+
+diff --git a/re2/re2.h b/re2/re2.h
+index 32b2718..de66689 100644
+--- a/re2/re2.h
++++ b/re2/re2.h
+@@ -956,7 +956,7 @@ namespace hooks {
+ // of Apple platforms that aren't macOS. If an iOS application really needs
+ // the context pointee someday, we can get more specific then...
+ #define RE2_HAVE_THREAD_LOCAL
+-#if defined(__APPLE__) && !TARGET_OS_OSX
++#if defined(__APPLE__)
+ #undef RE2_HAVE_THREAD_LOCAL
+ #endif
+ 
